core/arch/arm/kernel/thread_spmc.c

```diff
@@ -1184,13 +1187,24 @@ static int handle_mem_share_tmem(paddr_t pbuf, size_t blen, size_t flen,
 
 	cpu_spin_lock(&rxtx->spinlock);
 	rc = spmc_read_mem_transaction(rxtx->ffa_vers, buf, flen, &mem_trans);
-	if (!rc && IS_ENABLED(CFG_NS_VIRTUALIZATION) &&
+	if (rc)
+		goto unlock;
+
+	if (is_sp_share(&mem_trans, buf)) {
+		rc = spmc_sp_add_share(&mem_trans, buf, blen, flen,
+				       global_handle, NULL);
+		goto unlock;
+	}
+
+	if (!IS_ENABLED(CFG_NS_VIRTUALIZATION) &&
```

Comment:
Shouldn't this be `if (IS_ENABLED(CFG_NS_VIRTUALIZATION) && virt_set_guest(mem_trans.sender_id))` ?

Reply:
You're right, I fixed this line.